### PR TITLE
EVA-1885 Clustering prototype

### PR DESCRIPTION
**Pull request by andresfsilva**
Working prototype for some simple cases for the clustering pipeline. 

**Reader**: Read from a VCF file and get Variants
**Processor**: Convert Variants into SubmittedVariantEntities (to write to mongo)
**Writer**: 
1. Search if the clustered variant hash already have assigned an RS accession, if not assign one and create the clustered variant in mongo (Now this is simulated, later will be done by the Accessioning Service)
2. Update the `rs` field for the submitted variants

Some things are not included in the current prototype:
- Use of `accessioning service` for creating RD ids (including writing to clustered variants collection)
- Current `VcfReader` **does not read the ID column**, we should either create our implementation or modify the current one in the `variation-commons` repository

For the not included items an implementation was included to simulate their behaviour

### CODE REVIEW

File: eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/processors/VariantToSubmittedVariantEntityProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Test
+    public void process() {
+        Variant variant = new Variant("1", 1000L, 1000L, "A", "T");
+        SubmittedVariantEntity processedVariant = processor.process(variant);
+
+        SubmittedVariant submittedVariant = new SubmittedVariant(ASSEMBLY_ACCESSION, 0, "", "1", 1000L, "A", "T", null);
+        String hash = hashingFunction.apply(submittedVariant);
+        SubmittedVariantEntity expectedSubmittedVariantEntity = new SubmittedVariantEntity(1L, hash, submittedVariant, 1);
```
Comment (sundarvenkata-EBI) [inline eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/processors/VariantToSubmittedVariantEntityProcessorTest.java]: I think this should be:
```suggestion
        SubmittedVariantEntity expectedSubmittedVariantEntity = new SubmittedVariantEntity(1000L, hash, submittedVariant, 1);
```
But SVE equals method doesn't care about the accession, so the test still passes...

File: eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/SubmittedVariantWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    public void tearDown() {
+        mongoTemplate.dropCollection(SubmittedVariantEntity.class);
+    }
+
+    @Test
+    @UsingDataSet(locations = {"/test-data/submittedVariantEntity.json"})
+    public void writer() throws Exception {
+        List<SubmittedVariantEntity> submittedVariantEntities = createSubmittedVariantEntities();
+        submittedVariantWriter.write(submittedVariantEntities);
+        assertTrue(allClustered());
```
Comment (sundarvenkata-EBI) [inline eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/SubmittedVariantWriterTest.java]: Another test to check the RS attribute for the documents?

File: eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/configuration/batch/io/SubmittedVariantWriterConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import static uk.ac.ebi.eva.accession.clustering.configuration.BeanNames.CLUSTERED_SUBMITTED_VARIANTS_WRITER;
+
+@Configuration
+@Import({MongoConfiguration.class})
+public class SubmittedVariantWriterConfiguration {
+
+    @Autowired
+    InputParameters inputParameters;
+
+    @Bean(CLUSTERED_SUBMITTED_VARIANTS_WRITER)
```
Comment (jmmut) [inline eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/configuration/batch/io/SubmittedVariantWriterConfiguration.java]: the bean is "clustered..." and the classes is "submitted...". Either now or later, rename to something more appropriate

File: eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/processors/VariantToSubmittedVariantEntityProcessor.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private String assemblyAccession;
+
+    private long submittedVariantAccession;
+
+    private Function<ISubmittedVariant, String> hashingFunction;
+
+    public VariantToSubmittedVariantEntityProcessor(String assemblyAccession) {
+        this.assemblyAccession = assemblyAccession;
+        hashingFunction = new SubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
+        //To simulate the VcfReader is reading the ID column
```
Comment (jmmut) [inline eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/processors/VariantToSubmittedVariantEntityProcessor.java]: I really hope we don't miss any of these mock code. I would add `// TODO`s everywhere to mark explicitly what should not stay.

Comment (tcezard) [reply eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/processors/VariantToSubmittedVariantEntityProcessor.java]: I agree with this however make sure you link the TODO to EVA-1897

Comment (tcezard) [reply eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/processors/VariantToSubmittedVariantEntityProcessor.java]: I just saw that you have already. Disregard

**Pull request merged**